src/chart/Treemap.tsx

```diff
@@ -868,6 +868,7 @@ export class Treemap extends PureComponent<Props, State> {
     const { width, height, className, style, children, type, ...others } = this.props;
     const attrs = filterProps(others, false);
     const viewBox = { x: 0, y: 0, width, height };
+    const ttContext = this.getTooltipContext();
```

Comment:
Treemap is one of the easier ones (only allows interaction with item, has no axes), I encourage you to pick up Line or RadialBar first. Once we have those covered we will know that solution will also work with Treemap. But not the other way around.

Reply:
This PR swaps to redux for all categorical charts and treemap.

Or at least it swaps what is currently being set from the generator in context set in redux. Not necessarily end state